src/util/ui.cpp

```diff
@@ -152,4 +152,38 @@ namespace util::ui {
 
         return nullptr;
     }
+
+    BadgeLayout createLayout(std::string _badgePNG, float _scale, cocos2d::CCPoint _pos, std::string _id, cocos2d::CCNode* _badgeWrapper) {
+        BadgeLayout layout;
+        layout.badgePNG = _badgePNG;
+        layout.scale = _scale;
+        layout.pos = _pos;
+        layout.id = _id;
+        layout.badgeWrapper = _badgeWrapper;
+        return layout;
+    }
+
+    uibuilder::Build<cocos2d::CCSprite> createBadge(BadgeLayout layout) {
+        log::info("creating badge");
+
+        auto badgeSprite = Build<cocos2d::CCSprite>::createSpriteName(layout.badgePNG.c_str())
+            .scale(layout.scale)
+            .pos(layout.pos)
+            .id(layout.id);
+
+        if (layout.badgeWrapper == nullptr) {
+            log::info("badge wrapper is nullptr");
+            return badgeSprite;
+        }
+        layout.badgeWrapper->updateLayout();
+
+        return badgeSprite;
+    }
+
+    uibuilder::Build<cocos2d::CCSprite> createBadgeIfSpecial(ccColor3B color, BadgeLayout layout) {
+        if (color == ccc3(15, 239, 195)) return createBadge(createLayout("role-mod.png"_spr, 1.f, layout.pos, "globed-mod-badge", layout.badgeWrapper));
+        if (color == ccc3(233, 30, 99)) return createBadge(createLayout("role-admin.png"_spr, 1.f, layout.pos, "globed-admin-badge", layout.badgeWrapper));
+        if (color == ccc3(154, 88, 255)) return createBadge(createLayout("role-supporter.png"_spr, 1.f, layout.pos, "globed-supporter-badge", layout.badgeWrapper));
+        if (color == ccc3(248, 0, 255)) return createBadge(createLayout("role-booster.png"_spr, 1.f, layout.pos, "globed-booster-badge", layout.badgeWrapper));
+    }
```

Comment:
can we stop ignoring compiler warnings. this is undefined behavior if the color didn't match anything because the function won't return a value.